lib/grape/middleware/formatter.rb

```diff
@@ -110,7 +110,7 @@ def read_rack_input(body)
           rescue Grape::Exceptions::Base => e
             raise e
           rescue StandardError => e
-            throw :error, status: 400, message: e.message
+            throw :error, status: 400, message: e.message, backtrace: e
```

Comment:
> Is this a standard pattern somehow in Ruby? I mean I would expect backtrace to be an actual backtrace and not the original exception. Is anyone doing it like this elsewhere?

I am quite new in the Ruby world, so TBH I don't know if this is a standard pattern but I would say no. However the Grape library as a "contract" for the formatter: `formatter.call(message, backtrace, options, env)`.

```ruby
      # lib/grape/middleware/error.rb:90
      def error_response(error = {})
        status = error[:status] || options[:default_status]
        message = error[:message] || options[:default_message]
        headers = { Grape::Http::Headers::CONTENT_TYPE => content_type }
        headers.merge!(error[:headers]) if error[:headers].is_a?(Hash)
        backtrace = error[:backtrace] || []
        rack_response(format_message(message, backtrace), status, headers)
      end

      # lib/grape/middleware/error.rb:103
      def format_message(message, backtrace)
        format = env[Grape::Env::API_FORMAT] || options[:format]
        formatter = Grape::ErrorFormatter.formatter_for(format, options)
        throw :error, status: 406, message: "The requested format '#{format}' is not supported." unless formatter
        formatter.call(message, backtrace, options, env)
      end
```

This method is called here:

```ruby
      # lib/grape/middleware/error.rb:29
      def call!(env)
        @env = env

        begin
          error_response(catch(:error) do
            return @app.call(@env)
          end)
        rescue StandardError => e
           # ...
        end
```

> Are there other places we throw :error like this that also need fixing?

Probably yes, but I didn't investing other use cases. This one in particulary affected us.